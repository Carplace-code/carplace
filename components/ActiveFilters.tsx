export interface ActiveFilter {
  label: string;
  onRemove: () => void;
}

interface ActiveFiltersProps {
  filters: ActiveFilter[];
  onClear: () => void;
}

export default function ActiveFilters({ filters, onClear }: ActiveFiltersProps) {
  return (
    <div className="mb-4 flex flex-wrap items-center gap-3 text-sm">
      {filters.map((f) => (
        <button
          key={f.label}
          type="button"
          className="flex items-center rounded-full bg-blue-100 px-3 py-1 text-blue-700"
          onClick={f.onRemove}
        >
          {f.label}
          <span className="ml-1 font-bold">&times;</span>
        </button>
      ))}
      {filters.length > 0 && (
        <button type="button" className="ml-2 text-sm text-red-500 underline" onClick={onClear}>
          Clear All
        </button>
      )}
    </div>
  );
}
