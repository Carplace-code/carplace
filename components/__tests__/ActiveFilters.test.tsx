import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

import ActiveFilters, { type ActiveFilter } from "@/components/ActiveFilters";

describe("ActiveFilters", () => {
  it("renders correctly with filters", () => {
    const onClear = vi.fn();
    const filters: ActiveFilter[] = [
      { label: "Brand: Toyota", onRemove: vi.fn() },
      { label: "Year: 2020", onRemove: vi.fn() },
    ];

    const { container } = render(<ActiveFilters filters={filters} onClear={onClear} />);
    expect(container).toMatchSnapshot();
  });

  it("renders nothing when no filters are present", () => {
    const onClear = vi.fn();
    const filters: ActiveFilter[] = [];

    const { container } = render(<ActiveFilters filters={filters} onClear={onClear} />);
    expect(container).toMatchSnapshot();
  });
});
