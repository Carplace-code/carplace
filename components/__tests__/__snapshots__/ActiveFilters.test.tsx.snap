// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActiveFilters > renders correctly with filters 1`] = `
<div>
  <div
    class="mb-4 flex flex-wrap items-center gap-3 text-sm"
  >
    <button
      class="flex items-center rounded-full bg-blue-100 px-3 py-1 text-blue-700"
      type="button"
    >
      Brand: Toyota
      <span
        class="ml-1 font-bold"
      >
        ×
      </span>
    </button>
    <button
      class="flex items-center rounded-full bg-blue-100 px-3 py-1 text-blue-700"
      type="button"
    >
      Year: 2020
      <span
        class="ml-1 font-bold"
      >
        ×
      </span>
    </button>
    <button
      class="ml-2 text-sm text-red-500 underline"
      type="button"
    >
      Clear All
    </button>
  </div>
</div>
`;

exports[`ActiveFilters > renders nothing when no filters are present 1`] = `
<div>
  <div
    class="mb-4 flex flex-wrap items-center gap-3 text-sm"
  />
</div>
`;
